vike/client/client-routing-runtime/prefetch/getPrefetchSettings.ts

```diff
@@ -109,3 +152,44 @@ function getPrefetchAttribute(linkTag: HTMLElement): PrefetchStaticAssets | null
 
   assert(false)
 }
+
+function getPrefetchPageContextAttribute(linkTag: HTMLElement): PrefetchPageContext | null {
+  const whenAttr = linkTag.getAttribute('data-prefetch-page-context-when')
+  const expireAttr = linkTag.getAttribute('data-prefetch-page-context-expire')
+
+  if (whenAttr === null && expireAttr === null) {
+    return null
+  }
+
+  if (whenAttr) {
+    if (whenAttr === 'false') {
+      return { when: false, expire: 0 }
+    }
+
+    if (whenAttr === 'hover') {
+      const correctValue: 'hover' = whenAttr.toLowerCase() as any
+
+      if (expireAttr === null) {
+        return { when: correctValue, expire: 0 }
+      }
+
+      if (!Number.isNaN(parseInt(expireAttr, 10))) {
+        return { when: correctValue, expire: parseInt(expireAttr, 10) }
+      }
+
+      assertUsage(false, `data-prefetch-page-context-expire has value "${expireAttr}" but it should instead be number`)
+    }
+
+    assertUsage(false, `data-prefetch-page-context has value "${whenAttr}" but it should instead be "false", "hover"`)
+  }
+
+  if (expireAttr !== null) {
+    if (!Number.isNaN(parseInt(expireAttr, 10))) {
+      return { when: 'hover', expire: parseInt(expireAttr, 10) }
+    }
+
+    assertUsage(false, `data-prefetch-page-context-expire has value "${expireAttr}" but it should instead be number`)
```

Comment:
There is some code repetition here. See code above which is a duplicate. This is client-side code so it's important to keep it compact.